flang/lib/Parser/prescan.cpp

```diff
@@ -1186,6 +1186,8 @@ Prescanner::IsFixedFormCompilerDirectiveLine(const char *start) const {
         break;
       }
       *sp++ = ToLowerCaseLetter(*p);
+    } else {
```

Comment:
If the loop is now going to terminate on a space, then the various `break` conditions in the loop can be combined to be less confusing.